apps/gesture_piarm/gesture_ssd_async_wout_robot.py

```diff
@@ -0,0 +1,233 @@
+            in_frame = in_frame.transpose((2, 0, 1))  # Change data layout from HWC to CHW
+            in_frame = in_frame.reshape((n, c, h, w))
+            exec_net.start_async(request_id=cur_request_id, inputs={input_blob: in_frame})
+        if exec_net.requests[cur_request_id].wait(-1) == 0:
+            inf_end = time.time()
+            det_time = inf_end - inf_start
+
+            # NOTICE: Parse detection results of the current request, only need one so choose max prob
+            proposals = []
+            res = exec_net.requests[cur_request_id].outputs[out_blob]
+            for obj in res[0][0]:
+                # Draw only objects when probability more than specified threshold
+                if obj[2] > args.prob_threshold:
+                    best_proposal = obj
+                    xmin = int(best_proposal[3] * initial_w)
+                    ymin = int(best_proposal[4] * initial_h)
+                    xmax = int(best_proposal[5] * initial_w)
+                    ymax = int(best_proposal[6] * initial_h)
+                    class_id = int(best_proposal[1])
+
+                    # Draw box and label\class_id
+                    color = (min(class_id * 12.5, 255), min(class_id * 7, 255), min(class_id * 5, 255))
+
+                    # NOTICE: Draw box based on center of original bounding box
+                    xmid = int(abs(xmin + xmax) / 2)
+                    ymid = int(abs(ymin + ymax) / 2)
+                    cv2.circle(frame, (xmid, ymid), 4, (0, 0, 255), thickness=-1, lineType=cv2.FILLED)
+                    cv2.rectangle(frame, (xmid - 100, ymid - 100), (xmid + 99, ymid + 99), color, 2)
+
+                    det_label = labels_map[class_id] if labels_map else str(class_id)
+                    cv2.putText(frame, det_label, (xmid, ymid - 7), cv2.FONT_HERSHEY_COMPLEX, 0.6, color, 1)
+                    proposals.append({'pred': class_id, 'pt1': (xmid - 100, ymid - 100), 'pt2': (xmid + 99, ymid + 99)})
+
+            # NOTICE: Robot control
+            if len(proposals) > 0:
+                robot_control(proposals, big_w, big_h)
+
+            # Regions for commands
+            cv2.rectangle(frame, (0, 50), (50, big_h - 50), (255, 255, 255), 2)
+            cv2.rectangle(frame, (50, 0), (big_w - 50, 50), (0, 0, 255), 2)
+            cv2.rectangle(frame, (50, big_h - 50), (big_w -50, big_h - 1), (255, 0, 0), 2)
+            cv2.rectangle(frame, (big_w - 50, 50), (big_w - 1, big_h - 50), (0, 255, 0), 2)
+
+            # NOTICE: Deletion of performance statsDraw performance stats
+
+        # NOTICE: Deletion of time stats
+        cv2.imshow("Detection Results", frame)
+
+        if is_async_mode:
+            cur_request_id, next_request_id = next_request_id, cur_request_id
+            frame = next_frame
+
+        key = cv2.waitKey(1)
+        if key & 0xFF == ord('q'): # NOTICE: Exit key is q, not CTRL + C
+            break
+        # NOTICE: Delection of option to do inference in synchronous mode
```

Comment:
deletion of option?

Reply:
The original Apache 2.0 code offered an option to switch between async and sync modes. I just wanted to get rid of this option since I wanted only the async part.

Reply:
was just commenting on the spelling of deletion: **delection** in the code comment

Reply:
oh embarrassing :(